staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go

```diff
@@ -78,10 +81,21 @@ type codec struct {
 	encodeVersion runtime.GroupVersioner
 	decodeVersion runtime.GroupVersioner
 
+	identifier runtime.Identifier
+
 	// originalSchemeName is optional, but when filled in it holds the name of the scheme from which this codec originates
 	originalSchemeName string
 }
 
+// identifier computes Identifier of Encoder based on codec parameters.
+func identifier(encodeGV runtime.GroupVersioner, encoder runtime.Encoder) runtime.Identifier {
+	identifier := fmt.Sprintf("encodeGV=%s", encodeGV)
```

Comment:
if we need the groupversioner state to be serializable, we have to make that part of the GroupVersioner interface :-/

out of curiosity, what does %s produce for the various GroupVersioner implementations?

Reply:
for those two types it works as expected, because:
- schema.GroupVersion has String() method [so that's what it returns]
- schema.GroupVersions is slice of a former, so it also works fine

There are couple more implementations (which doesn't seem to be use in any path we seem to be using) where it doesn't make that much sense (e.g. InternalGroupVersioner).

Let me add String() method to the interface and we can figure out better name later.

Reply:
Actually, I added Identifier() method.